src/components/list-item/list-item.tsx

```diff
@@ -147,44 +307,173 @@ export class ListItem implements ConditionalSlotComponent, InteractiveComponent
+          [CSS.hasCenterContent]: hasCenterContent,
+          [CSS.contentContainerDisabled]: disabled
         }}
-        disabled={disabled}
-        ref={(focusEl) => (this.focusEl = focusEl)}
-      >
-        {content}
-      </button>
-    ) : (
-      <div
-        class={{ [CSS.contentContainer]: true, [CSS.hasCenterContent]: hasCenterContent }}
-        ref={() => (this.focusEl = null)}
+        onClick={this.toggleSelected}
+        ref={(el) => (this.contentEl = el)}
+        role="gridcell"
       >
         {content}
-      </div>
+      </td>
     );
   }
 
   render(): VNode {
+    const {
+      openable,
+      open,
+      level,
+      setPosition,
+      setSize,
+      active,
+      label,
+      selected,
+      selectionAppearance,
+      selectionMode
+    } = this;
+
+    const showBorder = selectionMode !== "none" && selectionAppearance === "border";
+    const borderSelected = showBorder && selected;
+    const borderUnselected = showBorder && !selected;
+
     return (
-      <Host role="listitem">
-        <div class={CSS.container}>
+      <Host>
+        <tr
+          aria-expanded={openable ? toAriaBoolean(open) : null}
+          aria-label={label}
+          aria-level={level}
+          aria-posinset={setPosition}
+          aria-selected={toAriaBoolean(selected)}
+          aria-setsize={setSize}
+          class={{
+            [CSS.container]: true,
+            [CSS.containerBorderSelected]: borderSelected,
+            [CSS.containerBorderUnselected]: borderUnselected
+          }}
+          onFocus={this.focusCellNull}
+          onKeyDown={this.handleItemKeyDown}
+          ref={(el) => (this.containerEl = el)}
+          role="row"
+          tabIndex={active ? 0 : -1}
+        >
+          {this.renderSelected()}
+          {this.renderOpen()}
           {this.renderActionsStart()}
           {this.renderContentContainer()}
           {this.renderActionsEnd()}
-        </div>
-        <div class={CSS.nestedContainer}>
-          <slot />
+        </tr>
+        <div
+          class={{
+            [CSS.nestedContainer]: true,
+            [CSS.nestedContainerHidden]: openable ? !open : false
+          }}
+        >
+          <slot onSlotchange={this.handleDefaultSlotChange} />
         </div>
       </Host>
     );
   }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  handleDefaultSlotChange = (event: Event): void => {
+    const listItemChildren = getListItemChildren(event);
+    updateListItemChildren(listItemChildren);
+    const openable = !!listItemChildren.length;
+    this.openable = openable;
+    if (!openable) {
+      this.open = false;
+    }
+  };
+
+  toggleOpen = (): void => {
+    this.open = !this.open;
+  };
+
+  toggleSelected = (): void => {
+    if (this.disabled || this.selectionMode === "none") {
```

Comment:
`calcite-tree` emits the select event, but doesn't modify selection in `none` selection mode. Can we do the same here for consistency?